scipy/optimize/__init__.py

```diff
@@ -133,6 +133,7 @@
 
    nnls - Linear least-squares problem with non-negativity constraint.
    lsq_linear - Linear least-squares problem with bound constraints.
+   isotonic_regression - Least squares problem of isotonic regression via PAVA.
```

Comment:
One final point, is under the heading Linear least-squares the best place for this? I am not really familiar with the method so wanted to check

Reply:
I did not want to open a new section. Isotonic regression is minimizing squared error (=least squares) with some constraints for monotonicity. That's why I thought it might fit here.

Interestingly, one can replace the squared error by a huge class of other loss functions (e.g. the poisson deviance ~ neg poisson log-likelihood) and gets the same results. This is a point against the least squares section. But it is minor.